instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/semconv/url/UrlAttributesGetter.java

```diff
@@ -50,4 +50,14 @@ default String getUrlPath(REQUEST request) {
   default String getUrlQuery(REQUEST request) {
     return null;
   }
+
+  /**
+   * Returns the template used to build the full URL (if available)
+   *
+   * <p>Examples: {@code /users/{id}}; {@code /users?q={query}}
+   */
+  @Nullable
+  default String getUrlTemplate(REQUEST request) {
```

Comment:
Since `instrumentation-api` module is marked as stable we can not introduce unstable features here. You will have to add this to `instrumentation-api-incubator` module.